--- openstacksdk-0.39.0/openstack/orchestration/util/template_utils.py.orig	2019-12-09 10:52:37.000000000 +0100
+++ openstacksdk-0.39.0/openstack/orchestration/util/template_utils.py	2024-02-14 20:13:01.732527802 +0100
@@ -12,9 +12,9 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-import collections
 import json
 import six
+from six.moves import collections_abc
 from six.moves.urllib import parse
 from six.moves.urllib import request
 
@@ -151,7 +151,7 @@ def deep_update(old, new):
         old = {}
 
     for k, v in new.items():
-        if isinstance(v, collections.Mapping):
+        if isinstance(v, collections_abc.Mapping):
             r = deep_update(old.get(k, {}), v)
             old[k] = r
         else:
--- openstacksdk-0.39.0/openstack/resource.py.orig	2019-12-09 10:52:37.000000000 +0100
+++ openstacksdk-0.39.0/openstack/resource.py	2024-02-14 20:13:42.505640249 +0100
@@ -31,7 +31,6 @@ converted into this Resource class' appr
 and then returned to the caller.
 """
 
-import collections
 import inspect
 import itertools
 
@@ -42,6 +41,7 @@ from keystoneauth1 import discover
 import munch
 from requests import structures
 import six
+from six.moves import collections_abc
 
 from openstack import _log
 from openstack import exceptions
@@ -207,7 +207,7 @@ class Computed(_BaseComponent):
     key = "_computed"
 
 
-class _ComponentManager(collections.MutableMapping):
+class _ComponentManager(collections_abc.MutableMapping):
     """Storage of a component type"""
 
     def __init__(self, attributes=None, synchronized=False):
